platform/mds/mds/src/main/java/org/motechproject/mds/annotations/internal/EntityExtensionProcessor.java

```diff
@@ -0,0 +1,175 @@
+package org.motechproject.mds.annotations.internal;
+
+
+import org.motechproject.mds.annotations.EntityExtension;
+import org.motechproject.mds.annotations.Entity;
+import org.motechproject.mds.dto.EntityDto;
+import org.motechproject.mds.dto.FieldDto;
+import org.motechproject.mds.dto.RestOptionsDto;
+import org.motechproject.mds.dto.TrackingDto;
+import org.motechproject.mds.exception.entity.EntityDoesNotExtendEntityException;
+import org.motechproject.mds.exception.entity.EntityDoesNotExtendMDSEntityException;
+import org.motechproject.mds.javassist.MotechClassPool;
+import org.motechproject.mds.reflections.ReflectionsUtil;
+import org.osgi.framework.Bundle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.annotation.AnnotationUtils;
+import org.springframework.stereotype.Component;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.AnnotatedElement;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+import static org.motechproject.mds.util.Constants.AnnotationFields.HISTORY;
+import static org.motechproject.mds.util.Constants.AnnotationFields.NON_EDITABLE;
+
+/**
+ * The <code>EntityExtensionProcessor</code> provides a mechanism, allowing extending existing Entities.
+ *
+ * @see org.motechproject.mds.annotations.EntityExtension
+ */
+@Component
+public class EntityExtensionProcessor extends EntityAnnotationProcessor<EntityExtension> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(EntityExtensionProcessor.class);
+
+    private List<MDSProcessorOutput> entitiesProcessingResult;
+    private EntityProcessorOutput extendedEntityProcessorOutput;
+    private Set<Bundle> affectedBundles = new TreeSet<>();
+    private FieldProcessor fieldProcessor;
+
+
+    private boolean isClassValid(AnnotatedElement element) {
+
+        Class clazz = (Class) element;
+        Class<EntityExtension> extAnn = ReflectionsUtil.getAnnotationClass(clazz, EntityExtension.class);
+        Annotation annotationExtension = AnnotationUtils.findAnnotation(clazz, extAnn);
+
+
+        if (clazz.getSuperclass().getClass().getName().equals(Object.class.getName())) {
+            LOGGER.error("class {} annotated by EntityExtension does not extend any class", clazz.getName());
```

Comment:
Log and throw anti-pattern. Remove the error logging as we are throwing error with exact same message.